scripts/travisci/check_precommit.sh

```diff
@@ -1,15 +1,8 @@
 #!/usr/bin/env bash
 if [[ "${TRAVIS_PULL_REQUEST}" != "false" && "${TRAVIS}" == "true" ]]; then
-  files_changed=$(git diff "${TRAVIS_COMMIT_RANGE}" --name-only \
-    | grep ".*\.py")
-  if [[ ! -z "${files_changed}" ]]; then
-    pre-commit run --files  ${files_changed}
-  fi
+  pre-commit run --source ${TRAVIS_COMMIT_RANGE%...*} --origin HEAD
```

Comment:
source is the newest commit considered
origin is usually assumed to be older than source (HEAD should be newest)

the format of TRAVIS_COMMIT_RANGE is `SOURCE...ORIGIN`

splitting the TRAVIS_COMMIT_RANGE is IMO the right strategy. this only splits out the SOURCE part, but you should split both (i.e. don't assume ORIGIN==HEAD, though that's usually true).

Reply:
See https://git-scm.com/docs/gitrevisions#_specifying_ranges for full spec

Reply:
(reading the git spec, the pre-commit documentation is a little misguided on how commit ranges work, but we should stick to their naming scheme).

Reply:
sgtm